airbyte-integrations/connectors/source-recharge/source_recharge/manifest.yaml

```diff
@@ -0,0 +1,269 @@
+    incremental_sync:
+      type: CustomIncrementalSync
+      class_name: source_recharge.components.datetime_based_cursor.RechargeDateTimeBasedCursor
+      cursor_field: "updated_at"
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S%z"
+      datetime_format: "%Y-%m-%dT%H:%M:%S%z"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+      start_time_option:
+        type: RequestOption
+        field_name: "updated_at_min"
+        inject_into: request_parameter
+
+  # FULL-REFRESH STREAMS
+  # COLLECTIONS
+  collections_stream:
+    description: >-
+      Collections Stream: https://developer.rechargepayments.com/2021-11/collections/collections_list
+    $ref: "#/definitions/base_modern_api_stream"
+    $parameters:
+      name: "collections"
+      data_path: "collections"
+  # METAFIELDS
+  metafields_stream:
+    description: >-
+      Metafields Stream: https://developer.rechargepayments.com/2021-11/metafields
+    $ref: "#/definitions/base_modern_api_stream"
+    retriever:
+      $ref: "#/definitions/retriever_metafields"
+    $parameters:
+      name: "metafields"
+      data_path: "metafields"
+  # PRODUCTS
+  products_stream:
+    description: >-
+      Products Stream: https://developer.rechargepayments.com/2021-11/products/products_list
+      Products endpoint has 422 error with 2021-11 API version
+    $ref: "#/definitions/base_deprecated_api_stream"
+    $parameters:
+      name: "products"
+      data_path: "products"
+  # SHOP
+  shop_stream:
+    description: >-
+      Shop Stream: https://developer.rechargepayments.com/v1-shopify?python#shop
+      Shop endpoint is not available in 2021-11 API version
+    $ref: "#/definitions/base_deprecated_api_stream"
+    retriever:
+      $ref: "#/definitions/retriever_api_deprecated"
+      record_selector:
+        $ref: "#/definitions/selector"
+        extractor:
+          type: DpathExtractor
+          field_path: []
+    primary_key: ["shop", "store"]
+    $parameters:
+      name: "shop"
+
+  # INCREMENTAL STREAMS
+  # ADDRESSES
+  addresses_stream:
+    description: >-
+      Addresses Stream: https://developer.rechargepayments.com/2021-11/addresses/list_addresses
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "addresses"
+      data_path: "addresses"
+  # CHARGES
+  charges_stream:
+    description: >-
+      Charges Stream: https://developer.rechargepayments.com/2021-11/charges/charge_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "charges"
+      data_path: "charges"
+  # CUSTOMERS
+  customers_stream:
+    description: >-
+      Customers Stream: https://developer.rechargepayments.com/2021-11/customers/customers_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "customers"
+      data_path: "customers"
+  # DISCOUNTS
+  discounts_stream:
```

Comment:
Differences observed:
* There is not slicing in the new implementation as `incremental_sync.step` is not defined. [The previous version explicitly set this as 30 days](https://github.com/airbytehq/airbyte/blob/2ce32ab24d0bbe8dfb848f907b2492d0d239f5d1/airbyte-integrations/connectors/source-recharge/source_recharge/api.py#L21)
* The low-code version is sorting using `sort_by=updated_at-asc`. I'm fine with that change